python_modules/dagster/dagster/_core/definitions/automation_tick_evaluation_context.py

```diff
@@ -259,16 +244,84 @@ def build_run_requests_from_asset_partitions(
     )
 
 
+def _build_backfill_requests(
+    entity_subsets: Sequence[EntitySubset[EntityKey]],
+    asset_graph: BaseAssetGraph,
+    run_tags: Optional[Mapping[str, str]],
+) -> Tuple[Sequence[RunRequest], Sequence[EntitySubset[EntityKey]]]:
+    """Determines a set of entity subsets that can be executed using a backfill.
+    If any entity subset has size greater than 1, then it and all assets connected
```

Comment:
what's your reasoning for having one backfill per connected subset of the asset graph vs creating a single backfill that has any asset with multiple partitions and all assets connected to them (even if the resulting backfill has N disconnected subsets)? i dont think the single backfill is necessarily a better approach, i'm mostly just curious if there were other reasons you had for making N backfills.

Reply:
My thinking was originally that unconnected subsets would be likely to be unrelated to each other, and so in the event that you need to cancel some amount of work, it's might be better to be able to individually cancel unconnected subsets.

Thinking through it a bit more though, the main case where you'd get multiple backfills on a single tick would likely be some sort of situation where you have runs kicking off exactly on cron ticks (as these are naturally synchronized with each other).

I could easily imagine a case where you have many root assets that have the same partitions def / cron condition, and launch at the same time, but perhaps don't have any immediate shared children.

Launching N separate backfills in that case would be bad.

So I think I've convinced myself that I should switch this to doing at most one backfill per tick.